packages/mui-material/src/TablePagination/TablePagination.js

```diff
@@ -240,15 +245,16 @@ const TablePagination = React.forwardRef(function TablePagination(inProps, ref)
         </TablePaginationDisplayedRows>
         <ActionsComponent
           className={classes.actions}
-          backIconButtonProps={backIconButtonProps}
+          backIconButtonProps={slotProps?.previousIconButton ?? backIconButtonProps}
           count={count}
-          nextIconButtonProps={nextIconButtonProps}
+          nextIconButtonProps={slotProps?.nextIconButton ?? nextIconButtonProps}
```

Comment:
We should refactor to using `slotProps` 😊

Reply:
@DiegoAndai here, we are using the corresponding keys from `slotProps` if specified. Otherwise, we default to using the deprecated `backIconButtonProps` & `nextIconButtonProps`. Would you suggest a different way of doing this?

Reply:
Good catch! Fixed in https://github.com/mui/material-ui/pull/39353/commits/7840337e5fdaec11a2dfdaed8cf4840a1e555dce